src/Hosting/TestHost/src/WebHostBuilderExtensions.cs

```diff
@@ -144,19 +158,45 @@ public static IWebHostBuilder UseSolutionRelativeContentRoot(
         this IWebHostBuilder builder,
         string solutionRelativePath,
         string applicationBasePath,
-        string solutionName = "*.sln")
+        string solutionName)
+    {
+        return builder.UseSolutionRelativeContentRoot(solutionRelativePath, applicationBasePath, [solutionName]);
+    }
+
+    /// <summary>
+    /// Sets the content root of relative to the <paramref name="solutionRelativePath" />.
+    /// </summary>
+    /// <param name="builder">The <see cref="IWebHostBuilder"/>.</param>
+    /// <param name="solutionRelativePath">The directory of the solution file.</param>
+    /// <param name="applicationBasePath">The root of the app's directory.</param>
+    /// <param name="solutionNames">The names of the solution files to make the content root relative to. If empty, defaults to *.sln and *.slnx.</param>
+    /// <returns>The <see cref="IWebHostBuilder"/>.</returns>
+    [SuppressMessage("ApiDesign", "RS0026:Do not add multiple public overloads with optional parameters", Justification = "Required to maintain compatibility")]
+    public static IWebHostBuilder UseSolutionRelativeContentRoot(
+        this IWebHostBuilder builder,
+        string solutionRelativePath,
+        string applicationBasePath,
+        ReadOnlySpan<string> solutionNames = default)
```

Comment:
Should this use `IReadOnlyCollection<string>` or similar instead? It's not exactly performance sensitive.